weave/trace_server/trace_server_interface.py

```diff
@@ -292,6 +292,20 @@ class TableCreateReq(BaseModel):
     table: TableSchemaForInsert
 
 
+class TableUpdateReq(BaseModel):
+    project_id: str
+    initial_digest: str
+    # insertions performed before popping so that it is easier for
+    # caller to know the index of insertion
+    insert_rows: typing.Optional[list[tuple[int, dict[str, typing.Any]]]] = None
```

Comment:
I think there's a subtle ambiguity here when we have multiple entries in `insert_rows`. Can we clearly specify and test the intended semantics on inserting multiple rows? I can see 3 interpretations:

1. Sequentially insert each row before row `n` of the table, updating indices as you go.
2. Simultaneously insert each row before row `n` of the original table, where multiple rows having the same insert index are inserted in order of the request.
3. Simultaneously insert each row before row `n` of the original table, where multiple rows having the same insert index are inserted in reverse order of the request.

Tables aside, consider the zero-indexed list `A B C` and `insert_rows` payload of `(1, D), (2, E), (2, F), (3, G)`.  Under each interpretation, the result would be:

1. Sequential: `A D F G E B C`
2. Simultaneous in-order `A D B E F C G`
3. Simultaneous reverse-order `A D B F E C G`

I modified the test in `test_weave_client` and verified it is currently consistent with the sequential interpretation, which does make OK sense if the intent is to mirror a series of calls to `list.insert(idx, ...)`

but I'd argue that either simultaneous interpretations could be equally valid.  The simultaneous interpretation makes it easier to make multiple edits without the knowledge of what other changes may be happening.

I don't have a strong preference but I think the tests should be explicit about this behavior

Reply:
Hey @nickpenaranda - thanks for the thoughtful comments. Per our discussion in slack, i revised the interface to be a sequential series of operations. This addresses this comment, as well as your other ones below.